### PR TITLE
Update UA and GA4 dropdowns behavior to only show once parent is specified

### PR DESCRIPTION
**Pull request by danielgent**
## Summary

Addresses issue #3243

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics-4/components/common/PropertySelect.test.js
Comment (danielgent) [inline assets/js/modules/analytics-4/components/common/PropertySelect.test.js:97]: in order to not get act errors need to do something as DOM is empty at the start and end of this test (so there's nothing to wait for in the DOM as per react-testing-library best practices), **AND** wp-data is doing lots of stuff

File: stories/module-analytics.stories.js
Comment (danielgent) [inline stories/module-analytics.stories.js:73]: Instead, should I make new static fixtures?  (they could get pretty big though!)

File: tests/e2e/specs/modules/analytics/setup-with-account-no-tag.test.js
Comment (danielgent) [inline tests/e2e/specs/modules/analytics/setup-with-account-no-tag.test.js:162]: Should this profile select be hidden? I assumed so (and then I would change this line as I did with line 150)

Is the account Id and profile Id being set automatically when the account is selected?

Comment (danielgent) [reply tests/e2e/specs/modules/analytics/setup-with-account-no-tag.test.js]: See here the interactive demo in storybook https://google.github.io/site-kit-wp/storybook/pull/3330/?path=/story/analytics-module--account-property-profile-select

File: stories/module-analytics.stories.js
Comment (eugene-manuilov) [reply stories/module-analytics.stories.js:73]: I think there is no need to create a new fixture for it. We just need to take and load the first three items from the accounts list. Then load some properties for the first two accounts of selected accounts. And some profiles for the first account to simulate all possible situations.

```javascript
const { accounts: allAccounts, properties, profiles } = fixtures.accountsPropertiesProfiles;
const accounts = allAccounts.slice( 0, 3 );

dispatch( STORE_NAME ).receiveGetAccounts( accounts );

dispatch( STORE_NAME ).receiveGetProperties( properties.map( ( property ) => { ...property, accountId: accounts[0].id } ), { accountID: accounts[ 0 ].id } );
dispatch( STORE_NAME ).receiveGetProfiles( profiles.map( ( profile ) => { ...profile, accountId: accounts[ 0 ].id } ), { accountID: accounts[ 0 ].id, propertyID: properties[ 0 ].id } );

dispatch( STORE_NAME ).receiveGetProperties( properties.map( ( property ) => { ...property, accountId: accounts[1].id } ), { accountID: accounts[ 1 ].id } );
dispatch( STORE_NAME ).receiveGetProfiles( [], { accountID: accounts[ 1 ].id, propertyID: properties[ 0 ].id } );

dispatch( STORE_NAME ).receiveGetProperties( [], { accountID: accounts[ 2 ].id } );
dispatch( STORE_NAME ).receiveGetProfiles( [], { accountID: accounts[ 2 ].id, propertyID: properties[ 0 ].id } );
```

I hope this will be sufficient to see the new functionality in the storybook :thinking:

File: assets/js/modules/analytics/components/common/PropertySelect.test.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/common/PropertySelect.test.js:119]: Once we set the accountID, we need to check that the PropertySelect element becomes visible.

File: assets/js/modules/analytics/components/common/ProfileSelect.test.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/common/ProfileSelect.test.js:154]: Same here, but we also need to set propertyID and check that ProfileSelect element becomes visible.

File: assets/js/modules/analytics/components/common/ProfileSelect.test.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/common/ProfileSelect.test.js:164]: Just combine it into one action:

```javascript
dispatch( STORE_NAME ).setSettings( { accountID } );
```

File: assets/js/modules/analytics/components/common/ProfileSelect.test.js
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/common/ProfileSelect.test.js:180]: We should set the `propertyID` instead, because the `accountID` setting is already set in the setup function. Plus, we need to check that the ProfileSelect element becomes visible after doing these changes.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @danielgent. It's a good start. Added some comments for you. Please, take a look.

File: assets/js/modules/analytics-4/components/common/PropertySelect.test.js
Comment (danielgent) [inline assets/js/modules/analytics-4/components/common/PropertySelect.test.js:95]: Just to note @eugene-manuilov that the changed functionality of using ` isValidAccountID` and `isValidPropertyID` results in this test changing

Just checking that's ok with you! :+1:

File: assets/js/modules/analytics/components/common/PropertySelect.test.js
Comment (danielgent) [reply assets/js/modules/analytics/components/common/PropertySelect.test.js]: So (as per my last comment) I'm reusing the existing tests but changing them to expect hidden

The current tests go from visible to hidden but I'll also dispatch extra actions to check it toggles back to visible :+1:

File: stories/module-analytics.stories.js
Comment (danielgent) [reply stories/module-analytics.stories.js]: I tried this out and it worked @eugene-manuilov  :exploding_head: 

I don't understand how though.  Surely when I select the third account on the list we should get errors as `wp-data` starts fetching properties for that account :thinking: 

Ah well, I must be missing something deeper in the code but I've already spent some time looking so I'll just trust you :smile:

**Comment by danielgent** [comment]
Due to these changes there is now an e2e test which fails when the assertion is both true *and* false (see last two commits)

**Comment by eugene-manuilov** [comment]
@googlebot I consent.

File: assets/js/modules/analytics/components/common/ProfileSelect.test.js
Comment (tofumatt) [inline assets/js/modules/analytics/components/common/ProfileSelect.test.js:153]:
```suggestion
		// A valid account ID was set, so the select should be visible.
```

File: assets/js/modules/analytics/components/common/ProfileSelect.test.js
Comment (tofumatt) [inline assets/js/modules/analytics/components/common/ProfileSelect.test.js:168]:
```suggestion
		// A valid accountID is provided, so the select component should not be disabled.
```

File: assets/js/modules/analytics/components/common/ProfileSelect.test.js
Comment (tofumatt) [inline assets/js/modules/analytics/components/common/ProfileSelect.test.js:180]:
```suggestion
		// After setting a valid property ID, the select should be visible.
```

File: assets/js/modules/analytics/components/common/PropertySelect.test.js
Comment (tofumatt) [inline assets/js/modules/analytics/components/common/PropertySelect.test.js:98]: Again: I know this wasn't your wording but let's take the time to simplify it:

```suggestion
	it( 'should not render if account ID is invalid', async () => {
```

File: assets/js/modules/analytics/components/common/PropertySelect.test.js
Comment (tofumatt) [inline assets/js/modules/analytics/components/common/PropertySelect.test.js:116]: Similar to the previous comment, let's clarify this and add at least what seems to be a missing noun 🙂

File: assets/js/modules/analytics/components/common/PropertySelect.test.js
Comment (tofumatt) [inline assets/js/modules/analytics/components/common/PropertySelect.test.js:121]:
```suggestion
		// A valid account ID was set, so the select should be visible.
```

File: stories/module-analytics-4.stories.js
Comment (tofumatt) [inline stories/module-analytics-4.stories.js:55]: Why does this need to be called `accountA`? It's the only account for this test, right?

```suggestion
			const account = {
				id: '1000',
				name: 'Account A',
			};
```

Same goes for `propertyA` below.

File: stories/module-analytics.stories.js
Comment (tofumatt) [inline stories/module-analytics.stories.js:89]: Here it makes sense to label these, but I find these single-letter differentiators a bit tough to scan.

Let's use more descriptive labels—`propertyOne` and `propertyTwo` works, or `firstProperty` and `secondProperty`. Equally `properties = [ {}, {} ]` would work 🙂 

I know it seems nitpicky, but these end up being equivalent to single-letter-variables like `e` or `a` because that's the only differentiator between the two variables, which isn't as easy-to-scan as my suggestions 😅

File: tests/e2e/specs/modules/analytics/setup-with-account-no-tag.test.js
Comment (tofumatt) [inline tests/e2e/specs/modules/analytics/setup-with-account-no-tag.test.js:187]:
```suggestion
					// Property dropdown should select "Set up a new property" option because there is no property associated with the current reference URL.
```

**Review by tofumatt — CHANGES_REQUESTED** [review]
Overall looks good and the story works well when I tried it out using the link, thanks! 👍🏻 

I think there's a bit of wording and test tweaks to do here before we merge this, but after that should be good. After you've taken a look through just assign me for another review and it should be good 👍🏻

File: assets/js/modules/analytics-4/components/common/PropertySelect.test.js
Comment (eugene-manuilov) [reply assets/js/modules/analytics-4/components/common/PropertySelect.test.js]: Updated.

File: assets/js/modules/analytics/components/common/PropertySelect.test.js
Comment (eugene-manuilov) [reply assets/js/modules/analytics/components/common/PropertySelect.test.js]: Updated.

File: stories/module-analytics-4.stories.js
Comment (eugene-manuilov) [reply stories/module-analytics-4.stories.js]: Updated.

File: stories/module-analytics.stories.js
Comment (eugene-manuilov) [reply stories/module-analytics.stories.js]: Updated.

**Review by tofumatt — APPROVED** [review]
Looks great 👍🏻

**Comment by tofumatt** [comment]
(Merging despite the Nightly E2E tests failing, as they appear unrelated—something to do with AMP + Nightly.)

**Pull request merged**
